core-backend/sandbox/src/env.rs

```diff
@@ -339,12 +342,19 @@ where
         }
     }
 
-    fn execute<F, T>(self, pre_execution_handler: F) -> EnvironmentExecutionResult<T, Self, EP>
+    fn execute<PrepareMemory, PrepareMemoryError>(
```

Comment:
Also, I'd leave `<F, E>`:
```suggestion
    fn execute<F, E>(
```